CORS support

robbiet480 opened the pull request:
**Description:**
This allows users to enable [CORS](https://en.wikipedia.org/wiki/Cross-origin_resource_sharing) by setting `cors: true` in the HTTP configuration.

The exact headers added if enabled are:

```
Access-Control-Allow-Origin: *
Access-Control-Allow-Headers: Origin, Accept, X-Requested-With, Content-type, X-HA-access
```

Currently, there is no support for adding custom headers as #933 requested. I'm not sure what the exact use case would be and would like to hear more from @SamMousa.

**Related issue (if applicable):** #933

**Example entry for `configuration.yaml` (if applicable):**

``` yaml
http:
  cors: true
  cors_allowed_origins:
    - robbie.io
  cors_allowed_headers:
    - Origin
    - Accept
```

**Checklist:**
If the code does not interact with devices:
- [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
- [ ] Tests have been added to verify that the new code works.

SamMousa commented:
I have no specific use case for custom headers; just thought it might be easy to add while you're add it.
I would recommending using a string setting for CORS though; since a user might want to set it to a specific domain(s) instead of *.

robbiet480 commented:
@SamMousa I looked into allowing the user to set a list of domains but found that CORS actually only supports either a *, a single domain, or "null" which is effectively a star. I don't believe there is a spec-compliant way to support multiple domains.

It's not hard to implement custom headers, I just wasn't sure what the use case was. I'd rather leave that functionality out until a time when it's needed.

SamMousa commented:
Not sure about multiple domains not allowed:
https://www.w3.org/TR/cors/#access-control-allow-origin-response-header

Regardless you could match the requests' origin headers against an array
value and return the appropriate one if found. (I have no use case that
requires it though)
On Apr 2, 2016 2:43 AM, "Robbie Trencheny" <notifications@github.com> wrote:

> @SamMousa <https://github.com/SamMousa> I looked into allowing the user
> to set a list of domains but found that CORS actually only supports either
> a *, a single domain, or "null" which is effectively a star. I don't
> believe there is a spec-compliant way to support multiple domains.
>
> It's not hard to implement custom headers, I just wasn't sure what the use
> case was. I'd rather leave that functionality out until a time when it's
> needed.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/balloob/home-assistant/pull/1680#issuecomment-204618367>
>

robbiet480 commented:
@SamMousa I hope you're happy that I had to do a whole **5 extra minutes of work** (!) to implement those features.

It was **REALLY HARD**.



No really though, I figured i'd just get it over with.

You can configure like this:

```yaml
http:
  cors: true
  cors_allowed_origins:
    - robbie.io
  cors_allowed_headers:
    - Origin
    - Accept
```

:cookie:

SEJeff commented:
Hi @robbiet480 nicely done, but would you mind adding documentation for this somewhere please?

robbiet480 commented:
@SEJeff I will once I get sign off on all the variable names and stuff

SEJeff commented:
Good man! That is likely a @balloob or @fabaff one.

balloob commented on homeassistant/components/http.py:
```diff
@@ -60,11 +66,19 @@ def setup(hass, config):
     development = str(conf.get(CONF_DEVELOPMENT, "")) == "1"
     ssl_certificate = conf.get(CONF_SSL_CERTIFICATE)
     ssl_key = conf.get(CONF_SSL_KEY)
+    allow_cors = conf.get(CONF_ALLOW_CORS, False)
```
```python
allow_cors = len(cors_origins) > 0
```

balloob commented on homeassistant/components/http.py:
```diff
@@ -60,11 +66,19 @@ def setup(hass, config):
+                                                HTTP_HEADER_X_REQUESTED_WITH,
+                                                HTTP_HEADER_CONTENT_TYPE,
+                                                HTTP_HEADER_HA_AUTH])
+    cors_origins = conf.get(CONF_CORS_ORIGINS, ['*'])
```
We should never default to match all. That's opening up security without the user being aware.

balloob commented on homeassistant/components/http.py:
```diff
@@ -60,11 +66,19 @@ def setup(hass, config):
     ssl_certificate = conf.get(CONF_SSL_CERTIFICATE)
     ssl_key = conf.get(CONF_SSL_KEY)
+    allow_cors = conf.get(CONF_ALLOW_CORS, False)
+    cors_headers = conf.get(CONF_CORS_HEADERS, [HTTP_HEADER_ORIGIN,
```
We should not allow configurable headers. It's our API and we know what is going on inside it.

Adding features for the sake of features is the worst thing to do. All of it will live in the code base _forever_. Every PR in the future will have to spin up an HTTP server to test functionality that no one might ever use.

balloob commented:
This will require tests and config validation to be written.

JshWright commented:
In my opinion, this sort of stuff is better suited to the Nginx (or whatever) instance you put in front of Home Assistant

SamMousa commented:
Well for advanced stuff maybe but my RPI has enough stuff to do without
running nginx.
On Apr 5, 2016 3:13 AM, "Josh Wright" <notifications@github.com> wrote:

> In my opinion, this sort of stuff is better suited to the Nginx (or
> whatever) instance you put in front of Home Assistant
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/balloob/home-assistant/pull/1680#issuecomment-205572486>
>

JshWright committed:
vol.IsFile() needs to be called

vol.IsFile results in an instance of the IsFile method being returned by
the schema validation process. It needs to be called in order to do the
right thing.

balloob committed:
Merge pull request #1739 from JshWright/fix/mqtt-config-validation

vol.IsFile() needs to be called

jaharkes committed:
Test function for mqtt topic validators and IsFile() linting fix (#1740)

* Add test function for mqtt topic validators.
* Fix for linting error on vol.IsFile()

jaharkes committed:
Be flexible in what we accept for script.delay configuration. (#1738)

Accept delay configuration even when someone forgets to indent the time

specification.



Also removed 'weeks' and 'microseconds' from acceptable delay values.

There is a new homeassistant release every 2 weeks and running scripts

are not persisting across restarts. And there is still the option of

using (weeks*7) days if the long delay is really necessary.



And if someone really depends on microsecond delay precision we are

unlikely to be able to provide this accuracy, even milliseconds is

suspect for that matter but will at least allow us to specify some

subsecond delay.

pavoni committed:
Handle Share off for events in OwnTracks app (#1737)

jaharkes committed:
Trying to fix failing sun component tests. (#1741)

Make sure we set a timezone to make UTC calculations predictable
and don't override the location of the test_home_assistant config.

balloob committed:
Add feature request link to issue template

balloob committed:
Update frontend (embed Roboto)

fabaff committed:
Remove print

pavoni committed:
Fix vera breaking component loading order. (#1749)

jaharkes committed:
Make tox run faster by using in-memory sqlite db. (#1752)

Two tests were invariably grinding my system to a halt as it was
applying migrations to an empty sqlite database. This patch reduces
the time of a pytest run from 50 to about 16 seconds on my desktop.

The difference is even more noticable on a slower laptop drive, but
is probably insignificant for all you fancy SSD users.

jaharkes committed:
Upgrade astral (#1754)

* Upgrade astral to 1.0

* Pass timezone as string, not object, to astral.Location.

jaharkes committed:
Add line numbers when we find duplicate keys. (#1750)

Danielhiversen committed:
Rfxtrx config validating (#1747)

* config validation

* Config validation of rfxtrx switch

* Config validation of rfxtrx light

* Config validation of rfxtrx

* Config validation of rfxtrx

balloob committed:
Build frontend

balloob committed:
Correct frontend version

balloob committed:
Version bump to 0.17

balloob committed:
Merge pull request #1727 from balloob/dev

0.17

balloob committed:
Version bump to 0.18.0.dev0

turbokongen committed:
We need to allow extra keys on top level componenet config

fixes #1756

fabaff committed:
Add comment about location of hass (fixes #1723)

balloob committed:
Merge pull request #1758 from turbokongen/patch-1

We need to allow extra keys on top level componenet config

jaharkes committed:
Fix for MQTT config validation on the protocol field. (#1765)

jaharkes committed:
Fix for light service validation. (#1770)

Incorrect validation tested if passed value was a list instead of
a member of the list.

balloob committed:
Update frontend with weblink fix

jaharkes committed:
Accept group without entities in configuration. (#1768)

* Accept group without entities in configuration.

People seem to use these as placeholders for future expansion of their
home automation dreams, and we used to accept them.  We still have to
specify at least one of 'name', 'view' or 'icon' so that the group is
parsed as a dictionary.

* Also accept empty entities: key in a group.

* Additional fix for empty entities value in a group config.

balloob committed:
0.17.1 (#1771)

* We need to allow extra keys on top level componenet config

fixes #1756

* Add comment about location of hass (fixes #1723)

* Fix for MQTT config validation on the protocol field. (#1765)

* Update frontend with weblink fix

* Fix for light service validation. (#1770)

Incorrect validation tested if passed value was a list instead of
a member of the list.

* Accept group without entities in configuration. (#1768)

* Accept group without entities in configuration.

People seem to use these as placeholders for future expansion of their
home automation dreams, and we used to accept them.  We still have to
specify at least one of 'name', 'view' or 'icon' so that the group is
parsed as a dictionary.

* Also accept empty entities: key in a group.

* Additional fix for empty entities value in a group config.

* Version bump to 0.17.1

balloob committed:
Prevent device tracker from creating invalid YAML (#1774)

balloob committed:
Prevent device tracker from creating invalid YAML (#1774)

jaharkes committed:
Make yaml config parser errors look less like bugs. (#1776)

Instead of nested tracebacks, show a simpler error message.

    Config directory: /home/user/.homeassistant
    ERROR:homeassistant.util.yaml:duplicate key: "script"
      in "/home/user/.homeassistant/configuration.yaml", line 95, column 0
      in "/home/user/.homeassistant/configuration.yaml", line 108, column 0

jaharkes committed:
Make yaml config parser errors look less like bugs. (#1776)

Instead of nested tracebacks, show a simpler error message.

    Config directory: /home/user/.homeassistant
    ERROR:homeassistant.util.yaml:duplicate key: "script"
      in "/home/user/.homeassistant/configuration.yaml", line 95, column 0
      in "/home/user/.homeassistant/configuration.yaml", line 108, column 0

balloob committed:
Version bump to 0.17.2

balloob committed:
Merge pull request #1778 from balloob/hotfix/0.17.2

Hotfix/0.17.2

balloob committed:
Update netdisco to 0.6.2

OpenDave15 committed:
Allow use of pynetgear 3.3 port parameter. (#1777)

* Allow use of pynetgear 3.3 port parameter.

* Fix lint problem

joelash committed:
mqtt garage door component (#1742)

balloob committed:
Update frontend

balloob committed:
Update frontend

JshWright committed:
Don't round values in Thermostat internal state (#1782)

Since all values coming out of the Thermostat component pass though the
_convert_for_display() method (which handles any necessary rounding),
there is no need to round values that only exist in the internal state
of the thermostat device. It serves no purpose and risks rounding
errors/precision loss.

Danielhiversen committed:
Config validation of yr sensor (#1767)

balloob committed:
Update frontend

balloob committed:
Make all entity_ids lowercase (#1786)

robbiet480 committed:
Break Zeroconf into its own component

robbiet480 committed:
Final clean up, flake8, pylint, change a variable name, remove unnecessary imports

robbiet480 committed:
Block zeroconf from tests

Danielhiversen committed:
Config validation for rfxtrx sensor (#1780)

robbiet480 committed:
Revert all http.py changes

robbiet480 committed:
Use hass.config.api instead of hass.http

bimbar committed:
Expanded homematic component with MAX! support via homegear (#1783)

* Expanded homematic component with MAX! support via homegear
Also multithreading fixes

* fixed tox errors

* incorporate changes suggested by balloob

* replaced HomematicConfig Container Class with namedtuple

* fixed lint errors

TheRealLink committed:
Thinkingcleaner support (#1784)

* added first implementation of thinking cleaner

* fix lock release

* fixed tox flaws

* updated coveragerc

* fixed lock

* changed update lock

* fixed codestyle

balloob committed:
Prevent device tracker error

robbiet480 committed:
UPnP port mapping component

robbiet480 committed:
Remove unnecessary disable=no-name-in-module

robbiet480 committed:
Fix api_password conditional and close zeroconf when we shut down

robbiet480 committed:
Merge remote-tracking branch 'upstream/dev' into http-zeroconf

robbiet480 committed:
Update netdisco dependency to 0.6.3

robbiet480 committed:
Update netdisco dependency to 0.6.4 (deja vu all over again!)

robbiet480 committed:
Merge pull request #1733 from robbiet480/http-zeroconf

ZeroConf component

robbiet480 committed:
Disable import-error on upnp

balloob committed:
Remove stale print

robbiet480 committed:
Merge remote-tracking branch 'upstream/dev' into upnp

robbiet480 committed:
Change needs_auth ZeroConf key

robbiet480 committed:
Merge pull request #1790 from robbiet480/upnp

UPnP port mapping component

robbiet480 committed:
Fix flake8 error on zeroconf.py

JshWright committed:
Use constant time comparison for http authentication (#1804)

In order to prevent a potential timing attack, it's important to make
sure the password check takes the same amount of time, regardless of
how many characters in the candidate password match the real password.

This commit does increase the verbosity of the authentication check.
Generally it is a good idea for authentication logic to be very clear,
even if that requires some extra verbosity.

jaharkes committed:
Do not propagate api password (#1797)

* Do not propagate API password in service requests.

It makes service validation fail. The choice is to either handle it as an
optional key in every service handler and make sure it doesn't end up in event
stream and notifications, or to strip it as early as possible.

* Some places still need a forwarded api password.

- Event forwarding/remote api uses the local api password to
  authenticate against the remote instance.
- The generated index.html at '/' embeds the api password.

jaharkes committed:
Do not propagate api password (#1797)

* Do not propagate API password in service requests.

It makes service validation fail. The choice is to either handle it as an
optional key in every service handler and make sure it doesn't end up in event
stream and notifications, or to strip it as early as possible.

* Some places still need a forwarded api password.

- Event forwarding/remote api uses the local api password to
  authenticate against the remote instance.
- The generated index.html at '/' embeds the api password.

balloob committed:
Version bump to 0.17.3

balloob committed:
Merge pull request #1807 from balloob/release-0.17.3

0.17.3

jaharkes committed:
Change local library path from {config_dir}/lib to {config_dir}/deps. (#1799)

Just on the off chance that someone who happens to run as root and also
doesn't correctly parse "just remove config /lib and restart".

jaharkes committed:
Service validation for the thermostat component.

jaharkes committed:
Service validation for switch component.

jaharkes committed:
Some fixes for yr config validation. (#1809)

The setup_platform function is trying to get CONF_LATITUDE and CONF_LONGITUDE,
but the validation schema was not accepting these.

Also moved CONF_MONITORED_CONDITIONS and CONF_ELEVATION to homeassistant.const
because they are used in other places.

jaharkes committed:
Nest config validation (#1810)

* Config validation for Nest platforms.

jaharkes committed:
Config and service validation for shell_command component.

fabaff committed:
Enable certificate verification and allow non-encrypted communication (#1763)

Commit:
Make owntracks.py pylint clean (#1808)

* refactor validation logic into validate_payload function
* add debugging messages to validation

jaharkes committed:
Service validation for script component.

jaharkes committed:
Service validation for scene component.

jaharkes committed:
Service validation for rollershutter component.

jaharkes committed:
Service validation for notify component.

balloob committed:
Update frontend

fabaff committed:
Update links

Danielhiversen committed:
Allow negative elevation in yr config

jaharkes committed:
New version of betamax breaks the yr.no sensor test.

Pin to betamax-0.5.1 for now.

jaharkes committed:
New version of betamax breaks the yr.no sensor test.

Pin to betamax-0.5.1 for now.

jaharkes committed:
Ignore tests/config/deps/ for both git and flake8.

Sometimes py.test leave some packages around in tests/config/deps.
Make sure these do not accidentally get pulled into a commit or
cause a local tox run to fail.

jaharkes committed:
Ignore tests/config/deps/ for both git and flake8.

Sometimes py.test leave some packages around in tests/config/deps.
Make sure these do not accidentally get pulled into a commit or
cause a local tox run to fail.

JshWright committed:
Include away temps in target temps

The target temperature methods were not taking the 'away' status
into account. leading to misleading target temps being reported.

JshWright committed:
Improve target temperature selection logic

When picking which of the high/low temperatures to display as the
"target", it makes more sense to take the outside temperature into
consideration, rather than the current temperature of the thermostat.

If the temperature outside is less than the temperature of the
thermostat, then we are far more likely to end up in "heating" mode
eventually, and vice versa, regardless of where the current inside
temp falls in the range between high and low.

JshWright committed:
Fix copy/paste logic error

jaharkes committed:
Service validation for logbook component.

jaharkes committed:
Service validation for lock component.

jaharkes committed:
Service validation for keyboard component.

jaharkes committed:
Service validation for input_slider component.

jaharkes committed:
Service validation for input_select component.

jaharkes committed:
Service validation for input_boolean component.

jaharkes committed:
Service validation for ifttt component.

jaharkes committed:
Service validation for garage_door component.

jaharkes committed:
Service validation for downloader component.

balloob committed:
Merge pull request #1811 from fabaff/moving

Update links

robbiet480 committed:
Fix links in README.rst

balloob committed:
Update submodule

robbiet480 committed:
Update pyGTFS version

fabaff committed:
Update file header and fix typos

jaharkes committed:
Service validation for conversation component.

jaharkes committed:
Service validation for browser component.

fabaff committed:
Merge pull request #1819 from fabaff/typos

Update file header and fix typos

jaharkes committed:
Service validation for alarm_control_panel component.

TheRealLink committed:
Yamaha AVR Input Select (#1796)

* added select input for yamaha

* set source_list to none on init

oeysteinhansen committed:
mysensors.py Prevent sensor name mixup (#1802)

Change the sensor name schema so nodes are not mixed up. The dot that was pressent before was ignored by home-assistant.
Before name:  <skeatchname> <node_id><child_id>  eg. wall 107
New name   :  <skeatchname> <node_id> <child_id>  eg. wall 10 7
This will result in a Home Assistant Entity ID of: sensor.wall_10_7

balloob committed:
Merge pull request #1816 from JshWright/feature/accurate-away-temps-nest

Improve Nest target temp display

infamy committed:
fix for plex not detecting plex players after power up (#1824)

balloob committed:
Merge pull request #1821 from home-assistant/service-validations

Service validations

balloob committed:
Update frontend + mdi

balloob committed:
Use standardised datetime format

MartinHjelmare committed:
Fix field type conflict in influxdb

* Add STATE_UNAVAILABLE to states that are ignored when writing to the
  database. This will avoid a field type error for string if the field
  already contains a different type, eg integer.
* Add test for ignored states for influxdb.
* Clean up influxdb tests.

balloob committed:
Update frontend

balloob committed:
Merge pull request #1833 from home-assistant/iso8601

Use standardised datetime format

icovada committed:
SCSGATE: Actually cycle through all devices to register (#1841)

* SCSGATE: Actually cycle through all devices to register

Modify _activate_next_device function to actually cycle through self-devices_to_register instead of simply checking whether there are any and onl registering one

* Deleted extra white lines

pavoni committed:
Add support for imperial meter and user soecified calorific value.

balloob committed:
Merge pull request #1838 from MartinHjelmare/fix-influxdb

Fix field type conflict in influxdb

deisi committed:
Fix for the issue #1323 (#1835)

Cinntax committed:
Allow mpd to handle the play_media service.  (#1831)

Even though there's not really a place in the UI for it, it will now be possible to set a playlist as part of a script, and ultimately a scene.

flavio committed:
Sonos: better handling of offline players (#1829)

When a sonos player goes offline an endless stream of
exceptions is raised. That happens because homeassistant keeps trying
to refresh its status.

This can happen even when a sonos player has gone offline **before**
homeassistant is started. The sonos players take some time before
realizing one of their mates is no longer online, leading to the same
stream of exceptions inside of homeassistant.

Three types of exceptions of can be raised:

  - `requests.packages.urllib3.exceptions.MaxRetryError`
  - `requests.packages.urllib3.exceptions.NewConnectionError`
  - `TimeoutError`

It's not possible to handle all of them with a single `except` block
because they are raised in a random order and after some delays. That
means a 2nd or 3rd exception can take place while handling the 1st one.

The only solution is to check whether a a player is actually reachable
by attempting to connect to a service that must be running on it.

Also all the players in a 'unknown' state should not be polled by
homeassistant (despite of their brand).

I'm going to upstream the `_is_reachable` method I added to the
`sonos.py` file into `SoCo`. In the meantime we must ship this piece
of code with homeassistant.

Signed-off-by: Flavio Castelli <fcastelli@suse.com>

balloob committed:
Update bootstrap frontend script

turbokongen committed:
Zwave import fixes for stability (#1845)

jpmossin committed:
Only add visible sonos devices

Some Sonos devices (e.g SUB) does not have
a upnp media renderer, but are discovered as sonos
devices. Creating a SonosDevice object from such a
device will fail.

jaharkes committed:
Bump pywemo version to 0.4.2

bradsk88 committed:
Upgrading to python-wink 0.7.4 and improving RGB color support in HA (#1832)

balloob committed:
Catch platform setup error (#1847)

especially-relative committed:
Do not setup or configure a hue bridge twice (#1848)

Tracks previously configured bridges by IP and prevents duplicates
from being configured

pavoni committed:
Merge pull request #1842 from home-assistant/add_imperial_meter

Add support for imperial gas meter and user specified gas calorific value.

Danielhiversen committed:
rfxtrx config validation

bimbar committed:
Support for EQ3 Bluetooth Smart Thermostats (#1839)

* Initial Support for EQ3 Bluetooth Smart Radiator Thermostats

* tox runs successfully

* Moved device specific stuff to bluepy_devices library

* lint fix

Commit:
Add feedreader component (#1836)

vmulpuru committed:
Add bluetooth device tracker (#1830)

This tracker discovers new devices on boot and tracks bluetooth devices
periodically based on interval_seconds value. Devices discovered are
stored with 'BT_' as the prefix for device mac.

Requires PyBluez

sch3m4 committed:
Set name to hyperion assets (#1822)

* set name to hyperion assets

by specifying the name in the configuration:

light kodi_xxx:
  name: "Kodi XXX"
  platform: hyperion
  host: 192.168.1.222
  port: 19444

* Update hyperion.py

fix extra space

balloob committed:
Add bluepy package dependency to Dockerfile

TheRealLink committed:
add media_player webos tv platform (#1853)

* added webos tv platform

* moved coverage entries

craftyjon committed:
Pass format through to temper_device so that Fahrenheit works (#1858)

balloob committed:
Correct celcius to celsius (#1860)

balloob committed:
Add mysensors tcp ethernet gateway (#1861)

* Bump version of pymysensors to 0.6, which includes the tcp gateway.
* Update requirements_all.txt.
* Replace CONF_PORT with CONF_DEVICE and ATTR_PORT with ATTR_DEVICE.
* Add tcp_port in config.
* Try to guess if tcp or serial gateway is configured, by validating
	device name as an ip address. If successful setup tcp gateway, if it
	fails, setup serial gateway.
* Update device_state_attributes to show correct device, ethernet or
	serial.

balloob committed:
Update frontend

balloob committed:
Sort .coveragerc

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

Conflicts:
	homeassistant/bootstrap.py
	homeassistant/components/frontend/version.py
	homeassistant/const.py

balloob committed:
Fix wrong merge

Christoph-Wagner committed:
Removed redundant Throttle decorator (#1862)

TheRealLink committed:
fixed some issues with webos (#1856)

* fixed some issues

* fixed linter

balloob committed:
Update frontend

balloob committed:
Version bump to 0.18

balloob committed:
Version bump to 0.19.0.dev0

balloob committed:
Update underline of header README

turbokongen committed:
Improve  recognition for zwave thermostats

especially-relative committed:
Update unifi version

New unifi version has a fix that will allow it to install correctly

Danielhiversen committed:
Tellstick config validation

Danielhiversen committed:
Heat control config validation

* heat control configuration validation

* fix heat contol test

balloob committed:
Allow variables in service.call_from_config

balloob committed:
Alexa: Expose intent variables to service calls

balloob committed:
Automation: Add trigger context and expose to action

balloob committed:
Add script logic into helper.

TheRealLink committed:
WebOS component fixes

* fixed some exceptions

* add requirements to notify

* added optimistic state to power off

* run requirements script

balloob committed:
Move script component tests to script helper tests

balloob committed:
Script helper: support variables

balloob committed:
Script entities to allow passing in variables

balloob committed:
Automation: Allow embedding script definition

balloob committed:
Allow calling scripts from Alexa

fabaff committed:
Add link to docs

persandstrom committed:
new version and changed "key"

persandstrom committed:
Merge pull request #1889 from persandstrom/verisure_updated_links

Verisure: new version and changed "key" in smartplug

balloob committed:
Show error message frontend if can't find server (#1886)

balloob committed:
Allow setting entity namespace (#1887)

balloob committed:
Cache script object for Alexa

balloob committed:
Service/Script cleanup

balloob committed:
Merge pull request #1882 from home-assistant/service-parameters

Trigger variables in automation actions

balloob committed:
Netdisco bump to 0.6.6

balloob committed:
Merge pull request #1892 from home-assistant/netdisco-update

Netdisco update

sander76 committed:
external dependency evohomeclient 0.2.4 gave a json decode error. (#1895)

* external dependency evohomeclient 0.2.4 gave a json decode error.
upped to 0.2.5

fabaff committed:
Upgrade slacker to 0.9.10

Commit:
New configuration for rfxtrx sensor

Commit:
Update rfxtrx tests to handle new config

ishults committed:
Use protocol of web page for yr.no pictures

SEJeff committed:
Merge pull request #1901 from ishults/dev

Use protocol of web page for yr.no pictures

SEJeff committed:
Merge pull request #1896 from fabaff/slacker-upgrade

Upgrade slacker to 0.9.10

deisi committed:
Ontime and delay attribute are allways present in deutsche_bahn sensor (#1875)

Bart274 committed:
Update pyicloud version (#1849)

* Update requirements_all.txt

* Update icloud.py

Commit:
Refactor rfxtrx code

turbokongen committed:
Exclude locally built python-openzwave from tox. (#1911)

Excludes `build` directory built by running build_python_openzwave script. When running tox test locally.

robbiet480 committed:
Dont load price sensors for metered (read: taxi/cabs) products

Cinntax committed:
Summary: Enhanced to make more robust and efficient. (#1917)

Prevented a switch from being turned on twice.

Made the module regex more robust.

Refactored the code to reduce the amount of network traffic to/from pulseaudio.

Fixed pylint issues

milaq committed:
Initial support for generic 433Mhz GPIO adapters on a Raspberry Pi (#1865)

* initial support for generic 433mhz gpio adapters

* rpi-rf: refactor id_on/id_off to code_on/code_off

fits the purpose better and improves understanding

* rpi-rf: use v0.9.4

* rpi-rf: update features and dependencies to v0.9.5

includes the ability to optionally specify a protocol for each switch

* rpi-rf: remove explicit RPi.GPIO dependency

already a dependency of the rpi-rf module

* rpi-rf: make setting gpio, code_on and code_off mandatory

* rpi-rf: remove unused value_template

* rpi-rf: only enable TX once if there are switches

especially-relative committed:
Make zwave component user sane defaults. (#1891)

Change the zwave config default path to be valid in all but rare
edge cases. This will let that config value be truely optional.

Also check that libopenzwave is installed and print a warrning if
it cannot be found pointing at the site.

happyleavesaoc committed:
update snapcast with source select (#1908)

* update snapcast

* fix id

blackdog70 committed:
Arduino: Fixed pin_mode digital input (#1909)

Danielhiversen committed:
Merge pull request #1905 from home-assistant/rfxtrx_sensor_fix

Rfxtrx: Simplified configuration

gwendalg committed:
mqtt: Fix logic when embedded and broker configs are present. (#1919)

Fix test to prevent early exit of mqtt init handler when
both embedded and broker configs are present.

Signed-off-by: Gwendal Grignou <gwendal@gmail.com>

JshWright committed:
Ignore .venv (#1923)

When using virtualfish ('virtualenvwrapper' for the fish shell), you
can create a .venv file in a directory that contains the name of a
virtualenv that will be activated automatically when you cd into that
directory. This is a good and useful thing, but since folks will have
different names for their virtualenvs, we should ignore this file.

That... and I'm probably the only one using fish/virtualfish...

kylehendricks committed:
Fix a couple dead links in the README (#1936)

fabaff committed:
Upgrade python-telegram-bot to 4.0.1 (#1927)

bradsk88 committed:
Upgrading to python-wink 0.7.5. (#1925)

This fixes a bug where light bulb statuses were failing to update on the HA polling update call.

fabaff committed:
Upgrade messagebird to 1.2.0 (#1928)

fabaff committed:
Upgrade schiene to 0.15 (#1929)

fabaff committed:
Upgrade TwitterAPI to 2.4.1 (#1931)

fabaff committed:
Upgrade jsonrpc-requests to 0.2 (#1937)

balloob committed:
Update ISSUE_TEMPLATE.md

TheRealLink committed:
Yamaha: added mapping and exclude lists (#1880)

* added mapping and exclude lists

* reworked pr

* made code more pythonic

balloob committed:
Initial script condition support (#1910)

fabaff committed:
Upgrade pytz to 2016.4 (#1932)

* Upgrade pytz to 2016.4

* Upgrade pytz to 2016.4

gwendalg committed:
script: Fix log output (#1918)

Signed-off-by: Gwendal Grignou <gwendal@gmail.com>

balloob committed:
Fix script merge conflict

Commit:
Improve feedreader roboustness (#1926)

* bugfix: ignore not existing property in feedreader item
* add info and debug data
* split logic in smaller and hopefully easier to understand functions

balloob committed:
Update frontend

molobrakos committed:
ELIQ Online platform cleanup (#1942)

robbiet480 committed:
Fix variable name

robbiet480 committed:
Merge pull request #1914 from robbiet480/uber-metered-fix

Uber: Dont load price sensors for metered (read: taxi/cabs) products

fabaff committed:
Upgrade pushbullet.py to 0.10.0 (#1939)

MartinHjelmare committed:
Refactor mysensors component

* Add MySensorsDeviceEntity class to hold the common attributes,
	properties and methods for mysensors entities.
* Inherit from MySensorsDeviceEntity class in binary_sensor, light,
	sensor and switch mysensors platforms.
* Remove not needed attribute and method for const in GatewayWrapper
	class. The const attribute is already set in the wrapped object.
* Clean up state property for mysensors sensor entities.
* Inherit from MySensorsLightRGB in MySensorsLightRGBW class.
* Remove use of get_component in mysensors component and platforms.
* Clean up update method in MySensorsDeviceEntity class.

fabaff committed:
Add support for sensor classes (#1950)

Commit:
Fixed slack component bug with getting the channel when sending a message (#1954)

timharton committed:
Mqtt client key auth (#1935)

* Made changes so that the mqtt configuration can accept client keys and certs for auth.

* Need to figure out how the broker_config works, it's failing tests

* Fixed it so all tests passed and ssl feature works for all brokers
which are't embedded

* Bring into line with pep8

* Added config validation which has allowed me to make the code simpler

dcrypt3d committed:
Add Z-Wave lock support

aceat64 committed:
Forecast.io: Added minutely, hourly, and daily summary (#1943)

* Forecast.io: Added minutely, hourly, and daily summary

* Resolve pylint issue (Too many instance attributes)

balloob committed:
Script cleanup (#1963)

kylehendricks committed:
Add Pioneer AVR media_player support (#1968)

balloob committed:
Fix script condition issue (Thanks @bart274) (#1972)

deisi committed:
Added a switch to control beamers from acer (#1913)

turbokongen committed:
Initial HVAC component + Z-Wave platform (#1912)

Commit:
Support for OctoPrint sensors (#1924)

Commit:
Limit number of processed entries by Feedreader (#1966)

* process only last 20 available entries to avoid bombing event
  bus when parsing huge feeds
* trigger first update only when HA has completed startup,
  allowing components to complete subscriptions to feedreader
  events
* quote url in logs for better readability

Commit:
Update yaml parser for handling environment variables (#1967)

balloob committed:
Merge branch 'pr/1969' into dev

Conflicts:
	homeassistant/components/zwave.py

persandstrom committed:
add reconnect retry and longer timeouts for verisure (#1944)

Commit:
Fix an unresolved merge conflict

robbiet480 committed:
Merge pull request #1973 from kylehendricks/fix-unresolved-merge-conflict

Fix an unresolved merge conflict

infamy committed:
Update README.rst (#1981)

Splitting dev and general chat links in the badges

fabaff committed:
Fix typo

fabaff committed:
UPdate link to docs

fabaff committed:
Update link to docs

fabaff committed:
Add dweet export component (#1818)

hmronline committed:
Add Media Player Stop command + Kodi support for it (#1960)

* Started adding Stop command to Kodi media player

* minor

* minor

* minor

* abstract-method fixed

kylehendricks committed:
[Pioneer AVR] Display the currently select input source as the media title (#1974)

fabaff committed:
Update links to docs and doc strings (#1994)

fabaff committed:
Upgrade to PyMata 2.12 (#1993)

robbiet480 committed:
Merge branch 'cors' of github.com:robbiet480/home-assistant into cors

robbiet480 committed:
Do not allow wildcard CORS origins

robbiet480 committed:
CORS is enabled if there is at least 1 CORS origin

robbiet480 committed:
Do not allow setting custom CORS headers

robbiet480 committed:
Add http.py config validation

robbiet480 committed:
flake8 & pylint fixes

robbiet480 committed:
Set default for CORS origins

robbiet480 commented:
This is good to go. We are down to just `cors_allowed_origins`, a list in the `http` configuration. If there are any entries in that list CORS will be activated.

balloob commented:
This branch needs to be cleaned up before I can merge it. 319 files changed as I can't see what changes are included in this PR.

robbiet480 commented:
Closed by lucky number #2000 because it's a cleaner PR

trymbill commented:
I'm trying to set Access-Control-Allow-Origin to a wildcard (*) to allow access from any origin but setting my config like below doesn't seem to do the trick:

```
http:
  cors_allowed_origins:
    - '*'
```

Any clues? Am I missing something?

robbiet480 commented:
@trymbill We specifically decided not to allow a wildcard origin. You will need to specify specific origins.
